tests/unit/test_client.py

```diff
@@ -993,34 +993,103 @@ def test_create_bucket_w_object_success(self):
         json_sent = http.request.call_args_list[0][1]["data"]
         self.assertEqual(json_expected, json.loads(json_sent))
 
-    def test_download_blob_to_file_with_blob(self):
-        project = "PROJECT"
+    def test_download_blob_to_file_with_failure(self):
+        import requests
+        from google.resumable_media import InvalidResponse
+        from google.cloud import exceptions
+        from google.cloud.storage.blob import Blob
+
+        raw_response = requests.Response()
+        raw_response.status_code = http_client.NOT_FOUND
+        raw_request = requests.Request("GET", "http://example.com")
+        raw_response.request = raw_request.prepare()
+        grmp_response = InvalidResponse(raw_response)
+
+        blob_name = "blob-name"
+        media_link = "http://test.invalid"
         credentials = _make_credentials()
-        client = self._make_one(project=project, credentials=credentials)
-        blob = mock.Mock()
+        client = self._make_one(credentials=credentials)
+        bucket = _Bucket(client)
```

Comment:
It doesn't seem like the hand-rolled `_Bucket` shim is useful here:  none of the attributes defined in that class get used.  Could it be just a `mock.Mock`?  Better yet, if we use an autospec mock for the `blob` (as I suggest below) we don't even need to create a bucket at all.